msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/CurveTool.java

```
@@ -28,117 +28,125 @@
+    return fwhm;
+  }
+
+  /**
+   * <p>
+   * roundFWHM method rounds the values of FWHM.This is Full width half maximum.
+   * 
+   * @param fwhm a {@link java.lang.Double} object.
+   * 
+   * @return roundedFWHM a {@link java.lang.Integer} object. This value is rounded integer value of
+   *         fwhm.
+   *         </p>
+   */
+  private int roundFWHM(double fwhm) {
+    int roundedFWHM = (int) Math.round(fwhm + 0.5);
+    return roundedFWHM;
+  }
+
+  /**
+   * <p>
+   * createDataFile method creates text file for Gaussian results.
+   * 
+   * @param datapoint a
+   *        {@link io.github.msdk.featdet.ADAP3D.common.algorithms.SliceSparseMatrix.VerticalSliceDataPoint}
+   *        list. This list contains m/z values and corresponding intensities.
+   * @param parameters a {@link java.lang.Double} array. This is array of Gaussian parameters.
+   * @param scanNumber a {@link java.lang.Integer} object. This is random scan numbers for which
+   *        we're finding gaussian fit.
+   *        </p>
+   */
+  private void createDataFile(List<SliceSparseMatrix.VerticalSliceDataPoint> datapoint,
+      double[] parameters, int scanNumber) {
+    double gaussianY[] = new double[datapoint.size()];
+    String filename = "output" + scanNumber + ".txt";
+    try {
+      File outputFile = new File(TEST_DATA_PATH + filename);
+      StringBuffer buffer = new StringBuffer();
+      FileWriter writer = new FileWriter(outputFile);
+
+      for (int i = 0; i < datapoint.size(); i++) {
+        gaussianY[i] = parameters[0] * Math.exp(
+            -Math.pow((datapoint.get(i).mz - parameters[1]), 2) / (2 * Math.pow(parameters[2], 2)));
+        buffer.append(datapoint.get(i).mz).append(",").append(datapoint.get(i).intensity)
+            .append(",").append(gaussianY[i]).append("\r\n");
+        writer.write(buffer.toString());
+        buffer = new StringBuffer();
+      }
+      writer.close();
+    }
+
+    catch (Exception e) {
+
```

Do not swallow exceptions.